src/par_iter/internal.rs

```diff
@@ -7,14 +7,20 @@ use join;
 use super::IndexedParallelIterator;
 use super::len::*;
 
+pub trait ProducerContinuation<ITEM> {
+    type Output;
+    fn with_producer<'p, P>(self, producer: P, shared: &'p P::Shared) -> Self::Output
+        where P: Producer<'p, Item=ITEM>;
+}
+
```

Comment:
(I think I also called it `internal` because I don't really want people to implement their own parallel iterators yet, until the interface calms down a bit.)